application/operator/NGSI-LD-operator/WaterPumpFog/function.js

```diff
@@ -26,47 +35,35 @@ exports.handler = function(contextEntity, publish, query, subscribe)
 	for (var key in contextEntity) {
 		updateEntity[key] = contextEntity[key]
 	}
-	console.log(updateEntity)
+
 	var con_observation = ""
-	var cof_observation = ""
+
 	if("on_status" in contextEntity) {
 		var con_observation = contextEntity.on_status.observedAt
 	}
-	if (con_observation != "" && onHasCreated == "false") {
-		value = "off"
-		onHasCreated = "true"
-		createDate = con_observation.split("T")
-		console.log("=====initial timer has been created======")
-	} else if (con_observation != "" && onHasCreated == "true") {
-		date = con_observation.split("T")
-		initiateUpdate = "true"
-		value  = "off"
-	}
-	if (initiateUpdate == "true") {
-		//console.log(date)
-		//console.log(createDate)
-		console.log(createDate[0])
-		console.log(date[0])
-		if (createDate[0] != date[0]) {
-			console.log("date is not equal")
+	current_observation = con_observation
+	if (current_observation != previous_observation) { 
+		previous_observation = current_observation
+		if (con_observation != "" && onHasCreated == "false") {
+			console.log("=====initial timer has been created======")
+			value = "off"
+			onHasCreated = "true"
+			createDate = con_observation.split("T")
+			publishStatus = "true"
+		}
+
+		if (publishStatus == "true") {
+			for(var i = 0 ; i<60 ;i++) {
+				console.log("Wait for publish")
+				sleep(1000);	
```

Comment:
(suggestion Better to define a global variable and replace with hardcoded value so that single change will be reflected everywhere.)

Reply:
incorporated in another pull  request